src/pkg/retention/policy/lwp/models.go

```diff
@@ -32,14 +34,21 @@ type Metadata struct {
 	Rules []*rule.Metadata `json:"rules"`
 }
 
-// FromMap constructs the metadata struct from map
-func (meta *Metadata) FromMap(m map[string]interface{}) error {
-	mdata, err := json.Marshal(&m)
+// ToJSON marshals metadata to JSON string
+func (m *Metadata) ToJSON() (string, error) {
```

Comment:
I'm struggling to see why this is needed (along with the `FromJSON` pair below. What does this give us that calling `json.Marshal`/`json.Unmarshal` directly doesn't?

Reply:
code is not much or less, but it makes the calling easier and clear.